csrgen.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+import argparse
+import fnmatch
+import re
+import os
+import subprocess
+import readline
+from jinja2 import Template, Environment, FileSystemLoader
+
+file_loader = FileSystemLoader('.')
+env = Environment(loader=file_loader)
+
+
+description = """
+Simple script to handle generating Certificate Signing Requests (CSR)
+This has the ability to generate CSRs with and without a provided key.
+Also can handle multiple domains as Subject Alternative Name (SAN) records.
+"""
+config_template = """[ req ]
+    default_bits           = 2048
+    distinguished_name     = req_distinguished_name
+    prompt                 = no
+    req_extensions         = v3_req
+[ req_distinguished_name ]
+    {% if c is not none -%}
+        C                      = {{s}}
+    {%- endif %}
+    {%- if l is not none -%}
+        L                      = {{l}}
+    {%- endif %}
+    {%- if s is not none -%}
+        ST                     = {{s}}
+    {%- endif %}
+    {%- if o is not none -%}
+        O                      = {{o}}
+    {%- endif %}
+{{'    CN                     = {{ cn }}
+[ v3_req ]
+    subjectAltName          = @alt_names
+
+[alt_names]
+    DNS.1 = {{ cn }}
+    {%- for domain in sans %}
+        {%- if domain %}
+    DNS.{{ loop.index +1 }} = {{ domain }}
+        {%- endif %}
+    {%- endfor %}
+'}}
+"""
+
+
+def check_file(file):
+    try:
+        os.stat(file).st_size > 1
+    except:
+        return False
+    else:
+        return file
+
+
+def check_domain(input):
```

Comment:
Same comments as earlier:

- what are you checking for? consider `is_domain`
- what type is input?

Reply:
Two things. Check if it is a domain or if it is a file with domains. Maybe I should create 2 flags rather then combining them in one?